src/Compilers/Core/Portable/SourceGeneration/ISyntaxHelper.cs

```diff
@@ -43,6 +43,8 @@ internal interface ISyntaxHelper
 
     internal abstract class AbstractSyntaxHelper : ISyntaxHelper
     {
+        private static readonly ObjectPool<Stack<GreenNode>> s_nodeStackPool = new ObjectPool<Stack<GreenNode>>(static () => new Stack<GreenNode>());
```

Comment:
Consider just using an `ArrayBuilder`; they're already pooled, and we use them as stacks all over the place. We even have a nice `TryPop` extension for them :slightly_smiling_face:.

Reply:
Add and TryPop, name a more iconic duo :-D

Reply:
Will make those changes. Thanks!

Reply:
done.